Re-formulate docker instruction (#45)
AlexsLemonade/RNA-Seq-Exercises: docker-load.md

# Docker Container Set Up - from Flashdrive

If you have successfully installed [Docker](https://github.com/AlexsLemonade/training-modules/blob/master/docker-install/README.md),
and have [copied the files from the flash drive](flashdrive-instructions.md),
you are now ready for the last part of set up: setting up your Docker container.   

Setting up your Docker container as described by these steps, will
provide you with all the software and packages you need for this workshop.

## Part 1: Get the Docker image using a copy from the flash drives

1. Be sure that you have copied `ccdl_training_rnaseq.tar.gz` from the flash drive
as was instructed in [these steps](flashdrive-instructions.md). If you do not
have this file yet, follow [those instructions](flashdrive-instructions.md) before
attempting to complete these next steps.

### Next steps

> * [Mac OS X and Linux](#mac-os-x-and-linux)

Reviewer: ```suggestion
* [Mac OS X and Linux](#next-steps-for-mac-os-x-and-linux-users)
```

> * [Windows 10 Pro](#windows-10-pro)

Reviewer: ```suggestion
* [Windows 10 Pro](#next-steps-for-windows-10-pro-users)
```

```
REPOSITORY                         TAG                 IMAGE ID            CREATED             SIZE

> ccdl/training_rnaseq               2019-czi        22f2b4f05051        3 days ago          5.28GB

Reviewer: Can you remove this image id in a way that indicates it needs to be updated by us at a later point?

Author: Yes. I will do that.

```
_Note that the created field may not match._

## Part 2: Run the Docker container (all operating systems)

1. Run the container. Change the `<PASSWORD>` in the line below to whatever you'd
  like. Make sure to get rid of `<` and `>`. Also note that your chosen PASSWORD
  cannot have a `$`.
```
docker run -e PASSWORD=<PASSWORD> -p 8787:8787 ccdl/training_rnaseq:2019-czi
```

2. Open `Kitematic` - you should see an image running.

3. `Settings` > `Volumes` > Set local folder to `training-modules` that was

> transferred from the flash/hard drive, using the `CHANGE` button.

Reviewer: ```suggestion
transferred from the flash drive, using the `CHANGE` button.
```

  - In a *Windows* or *Mac* in Kitematic, go to the `Settings` > `Hostname/Ports`
    tab and click on the blue lettering.
![Folder](screenshots/all-01-network.png)

  - Alternatively, for a *Mac*, you can navigate to the RStudio window by typing
    `localhost:8787` in your web browser

5. Log into `RStudio`. The username will be 'rstudio' and the password will be
whatever you selected above (can also be accessed from the `Settings` >
`General` panel).

6. You should see a `kitematic/` folder in your `RStudio` Files panel. Click on it.
If you do not see the training modules folders in the kitematic folder, raise
your hand.

7. To test that everything is set up, copy and paste the follow command in the
R Studio `Console` window.

```
source("kitematic/ready_script.R")
```

![Ready](screenshots/ready_command.png)

8. Click `Enter`. You should receive an indication that you are successfully set
up. Keep this indicator handy for the instructors and helpers as they come
around to check.

### If Kitematic doesn't work:

If all else fails and Kitematic is not working for you, go to your `Terminal` or
`Command Prompt` (for Mac or Windows respectively) and type in the following, but

> replacing <PATH_TO_TRAINING_FOLDERS> bit with the absolute path to

Reviewer: ```suggestion
replace <PATH_TO_TRAINING_FOLDERS> with the absolute path to
```

> "main directory" that was transferred from the flash/hard drive.

Reviewer: ```suggestion
`training-modules` that was transferred from the flash drive.
```

Reviewer: Carry over these changes to the other set of instructions as well.

```
docker run -it --rm --mount type=volume,dst=/home/rstudio/kitematic,volume-driver=local,volume-opt=type=none,volume-opt=o=bind,volume-opt=device=<PATH_TO_TRAINING_FOLDERS> -e PASSWORD=<PASSWORD> -p 8787:8787 ccdl/training_rnaseq:2019-czi
```
After starting your container this way, you can get to the RStudio window in
a similar way as described above:
- In Mac, type: `localhost:8787` in your web browser.
- In Windows, go to `Command Prompt`, type: `ipconfig` and click enter.
  Find the number that corresponds to the `Virtual Box Host Network` and the
  `IPv4 Address`. Copy and paste it.
  Put that number and `:8787` at the end of it in your browser.

> Resume with step 6 and 7.

Reviewer: ```suggestion
Resume with step 5 of part 2.
```
I think?

Author: Looks like Step 4 of part 2 but point taken.  I think the other one is also wrong in this respect.

Author: Nope, never mind. You and past Candace were right.
